src/main/java/htsjdk/samtools/reference/FastaSequenceIndexCreator.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Daniel Gomez-Sanchez
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package htsjdk.samtools.reference;
+
+import htsjdk.samtools.SAMException;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.tribble.readers.AsciiLineReader;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Static methods for create an {@link FastaSequenceIndex}.
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public final class FastaSequenceIndexCreator {
+
+    // cannot be instantiated because it is an utility class
+    private FastaSequenceIndexCreator() {}
+
+    /**
+     * Creates a FASTA .fai index for the provided FASTA.
+     *
+     * @param fastaFile the file to build the index from.
+     *
+     * @throws SAMException if the fai file already exists or the file is malformed.
+     * @throws IOException  if an IO error occurs.
+     */
+    public static void create(final Path fastaFile) throws IOException {
+        // get the index to write the file in
+        final Path indexFile = ReferenceSequenceFileFactory.getFastaIndexFileName(fastaFile);
+        if (Files.exists(indexFile)) {
+            // throw an exception if the file already exists
+            throw new SAMException("Index file " + indexFile + " already exists for " + fastaFile);
+        }
+        // build the index
+        final FastaSequenceIndex index = buildFromFasta(fastaFile);
+        index.write(indexFile);
+    }
+
+    /**
+     * Builds a FastaSequenceIndex on the fly from a FASTA file.
+     *
+     * Note: this also alows to create an index for a compressed file, but does not generate the
```

Comment:
I added this note because it is a difference with respect to the `samtools faidx` command.

I can also imagine scenarios where where a compressed fasta file is already indexed with bgzip (`.gzi` index), and the only requirement is to create the `.fai` index. Anyway, if there is a way to create with htsjdk a `.gzi` index I can integrate it here in the case of block-compressed input.